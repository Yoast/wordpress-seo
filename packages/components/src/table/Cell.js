import PropTypes from "prop-types";
import styled from "styled-components";

/* Yoast dependencies */
import { colors } from "@yoast/style-guide";

/**
 * Separatifies an element.
 *
 * @returns {String} CSS
 */
export function separatify() {
	return `
		&::after {
			position:relative;
			display: inline-block;
			border-right: 2px solid ${ colors.$color_grey };
			padding-right: 40px;
			height: 60px;
			content: "";
		}

		@media screen and ( max-width: 1355px ) {
			&::after {
				padding-right: 24px;
			}
		}

		@media screen and ( max-width: 800px ) {
			&::after {
				height: 48px;
				padding-right: 16px;
			}
		}
	`;
}

/**
 * Make an element text nowrap and add ellipsis.
 *
 * @returns {String} CSS
 */
export function ellipsify() {
	return `
		overflow: hidden;
		text-overflow: ellipsis;
		white-space: nowrap;
	`;
}

/*
 * The "Cells" are flex-items, their "flex" property is the initial `0 1 auto`
 * that translates to: cannot grow, can shrink, initial width based on the content.
 * A cell can be changed on a case by case basis wrapping the cells in a
 * styled component.
 */
const CellBase = styled.span`
	font-size: 14px;
	padding-left: 40px;

	&:first-child {
		padding-left: 0;
	}

	&::before {
 		position: absolute;
 		left: -9999em;
 		top: -30px;
 		font-size: 1.286em;
 		line-height: 0;
 		${ props => props.headerLabel ? `content: "${ props.headerLabel }";` : "content: none;" }
 	}

 	${ props => props.separator ? separatify() : "" }
	${ props => props.ellipsis ? ellipsify() : "" }

	@media screen and ( max-width: 1355px ) {
		padding-left: 20px;
		${ props => props.hideOnTablet ? "display: none;" : "" }
	}

	@media screen and ( max-width: 800px ) {
		${ props => props.hideOnMobile ? "display: none;" : "" }
	}
`;

CellBase.propTypes = {
	children: PropTypes.any,
	hideOnMobile: PropTypes.bool,
	hideOnTablet: PropTypes.bool,
	separator: PropTypes.bool,
	headerLabel: PropTypes.string,
	ellipsis: PropTypes.bool,
};

CellBase.defaultProps = {
	hideOnMobile: false,
	hideOnTable: false,
	separator: false,
	ellipsis: false,
};

/*
 * Primary cell, the largest one in a row: can grow, cannot shrink, and the
 * initial width is 200 pixels. In the responsive view, can shrink.
 */
export const CellPrimary = styled( CellBase )`
	flex: 1 0 200px;

	@media screen and ( max-width: 800px ) {
		flex-shrink: 1;
	}
`;

/*
 * Cell with fixed width: cannot grow, cannot shrink, and the width based on
 * its content.
 */
export const CellFixedWidth = styled( CellBase )`
	flex: 0 0 auto;
`;

/*
 * Cell with a minimum width: can grow, cannot shrink, and the initial width
 * is 100 pixels.
 */
export const CellMinWidth = styled( CellBase )`
	flex: 1 0 100px;
`;

/*
 * Cell with icon: cannot grow, cannot shrink, and the width is fixed based on
 * its content. The height is smaller in the responsive view because icons are
 * smaller.
 */
export const CellIcon = styled( CellFixedWidth )`
	height: 60px;

	@media screen and ( max-width: 800px ) {
		height: 48px;
	}
`;

/**
 * Makes cells use headers as bold labels with a colon in the mobile responsive view.
 *
 * @param {ReactElement} cell The original cell.
 *
 * @returns {ReactElement} The cell with transformed headers.
 */
export function responsiveHeaders( cell ) {
	return styled( cell )`
		@media screen and ( max-width: 800px ) {
			&::before {
				float: left;
				min-width: 60px;
				line-height: inherit;
				font-weight: 700;
				${ props => props.headerLabel ? `content: "${ props.headerLabel }:";` : "content: none;" }
			}

			> span {
				line-height: inherit;
			}
		}
	`;
}
