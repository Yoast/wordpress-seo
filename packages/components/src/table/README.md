# Usage guidelines for Table components
## Cells
- CellPrimary is meant to be the main cell in a row. There should be just one CellPrimary in each row.
- The responsiveHeaders function is meant to be used in combination with RowResponsiveWrap. This function ensures cells are full width in the responsive view.
- For the time being, headers should always be short and never break to a second line because they have `line-height: 0;`.
- CellIcon should only contain an icon, and therefore never use ellipsis.

