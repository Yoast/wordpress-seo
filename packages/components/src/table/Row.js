import PropTypes from "prop-types";
import styled from "styled-components";

/* Yoast dependencies */
import { colors } from "@yoast/style-guide";

// The Rows are flex containers.
export const Row = styled.li`
	background: ${ props => props.background };
	display: flex;
	min-height: ${ props => props.rowHeight };
	align-items: center;
	justify-content: space-between;
`;

Row.propTypes = {
	background: PropTypes.string,
	hasHeaderLabels: PropTypes.bool,
	rowHeight: PropTypes.string,
};

Row.defaultProps = {
	background: colors.$color_white,
	hasHeaderLabels: true,
};

/*
 * A row with responsive wrap allows children to wrap in new lines in the responsive view.
 * If the columns have headers, they're displayed as labels inside the column
 * content.
 */
export const RowResponsiveWrap = styled( Row )`
	@media screen and ( max-width: 800px ) {
		flex-wrap: wrap;
		align-items: flex-start;

		&:first-child {
			margin-top: ${ props => props.hasHeaderLabels ? "24px" : "0" };
		}

		// Use the column headers (if any) as labels.
		& > span::before {
			position: static;
			display: inline-block;
			padding-right: 0.5em;
			font-size: inherit;
		}
		& > span {
			padding-left: 0;
		}
	}
`;
