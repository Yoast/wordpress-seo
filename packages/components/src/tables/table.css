.yoast-table {
	font-size: var(--yoast-font-size-default);
	color: var(--yoast-color-default);
	line-height: 1.2;
	border-spacing: 0;
	border: var(--yoast-border-default);
	border-bottom: 0;
	width: 100%;
}

.yoast-table tbody tr:nth-child(odd) {
	background-color: #f9f9f9;
}

.yoast-table th {
	color: var(--yoast-color-dark);
	text-align: left;
	padding: 18px 12px;
	border-bottom: var(--yoast-border-default);
	white-space: nowrap;
	font-weight: var(--yoast-font-weight-bold);
}

.yoast-table td {
	padding: 18px 12px;
	border-bottom: var(--yoast-border-default);
}

.yoast-table th:first-child,
.yoast-table td:first-child {
	padding-left: 16px;
}

.yoast-table th:last-child,
.yoast-table td:last-child {
	padding-right: 16px;
}

td.yoast-table__image,
td.yoast-table__button {
	padding: 10px 18px 9px;
}

.yoast-table.yoast-table--nobreak td,
tr.yoast-table--nobreak td,
td.yoast-table--nobreak {
	white-space: nowrap;
}

th.yoast-table--primary {
	width: 100%;
}

td.yoast-table--nopadding {
	padding: 0 12px;
}
