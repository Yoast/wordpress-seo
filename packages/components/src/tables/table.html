<!--A button can both be used for a link and a button.
Use a link when the button should redirect to another page
use a button when it's used for an action: for example, opening a modal.-->

<!DOCTYPE html>
<html lang="en">

<head>

	<meta charset="utf-8">
	<meta name="viewport" content="width=device-width, initial-scale=1">

	<style>
		/* Demo styling */
		body {
			font-family: var(--yoast-font-family);
			font-size: var(--yoast-font-size-default);
			color: var(--yoast-color-font-default);
			line-height: 1.5;
		}
	</style>

	<link rel='stylesheet' href='../base/colors.css' type='text/css' media='all' />
	<link rel='stylesheet' href='../base/typography.css' type='text/css' media='all' />
	<link rel='stylesheet' href='../base/borders.css' type='text/css' media='all' />
	<link rel='stylesheet' href='../base/utility.css' type='text/css' media='all' />
	<link rel='stylesheet' href='../base/icons.css' type='text/css' media='all' />
	<link rel='stylesheet' href='../help-icon/help-icon.css' type='text/css' media='all' />
	<link rel='stylesheet' href='../button/buttons.css' type='text/css' media='all' />

	<link rel='stylesheet' href='table.css' type='text/css' media='all' />

</head>

<body class="yoast">

	<p>
		Standard table
	</p>

	<table class="yoast-table">
		<thead>
			<tr>
				<th>Related keyphrase</th>
				<th>
					Volume
					<a class="yoast-help" target="_blank" href="https://yoast.com">
						<span class="yoast-help__icon">
							<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 12 12" role="img" aria-hidden="true"
								focusable="false">
								<path
									d="M12 6A6 6 0 110 6a6 6 0 0112 0zM6.2 2C4.8 2 4 2.5 3.3 3.5l.1.4.8.7.4-.1c.5-.5.8-.9 1.4-.9.5 0 1.1.4 1.1.8s-.3.6-.7.9C5.8 5.6 5 6 5 7c0 .2.2.4.3.4h1.4L7 7c0-.8 2-.8 2-2.6C9 3 7.5 2 6.2 2zM6 8a1.1 1.1 0 100 2.2A1.1 1.1 0 006 8z" />
							</svg>
						</span>
						<span class="screen-reader-text">Tell what this link does.</span>
						<span class="screen-reader-text">(Opens in a new browser tab)</span>
					</a>
				</th>
				<th>Trend</th>
				<th>Action</th>
			</tr>
		</thead>
		<tbody>
			<tr>
				<td>yoast seo plugin</td>
				<td>800</td>
				<td class="yoast-table__image">
					<svg width="70" height="30" viewBox="0 0 70 30"><g><polygon fill="#ade3fc" points="0,28 0,12.36 6.363636363636364,12.36 12.727272727272728,7.32 19.09090909090909,12.36 25.454545454545457,7.32 31.818181818181817,7.32 38.18181818181818,7.32 44.54545454545455,2 50.909090909090914,2 57.27272727272727,7.32 63.63636363636363,2 70,2 70,28"></polygon><polyline fill="none" stroke="#498afc" stroke-width="2" points="0,12.36 6.363636363636364,12.36 12.727272727272728,7.32 19.09090909090909,12.36 25.454545454545457,7.32 31.818181818181817,7.32 38.18181818181818,7.32 44.54545454545455,2 50.909090909090914,2 57.27272727272727,7.32 63.63636363636363,2 70,2"></polyline></g></svg>
				</td>
				<td>
					<a href="#">Edit</a> / <a href="#">save</a>
				</td>
			</tr>
			<tr>
				<td>yoast seo plugin</td>
				<td>800</td>
				<td class="yoast-table__image">
					<svg width="70" height="30" viewBox="0 0 70 30"><g><polygon fill="#ade3fc" points="0,28 0,12.36 6.363636363636364,12.36 12.727272727272728,7.32 19.09090909090909,12.36 25.454545454545457,7.32 31.818181818181817,7.32 38.18181818181818,7.32 44.54545454545455,2 50.909090909090914,2 57.27272727272727,7.32 63.63636363636363,2 70,2 70,28"></polygon><polyline fill="none" stroke="#498afc" stroke-width="2" points="0,12.36 6.363636363636364,12.36 12.727272727272728,7.32 19.09090909090909,12.36 25.454545454545457,7.32 31.818181818181817,7.32 38.18181818181818,7.32 44.54545454545455,2 50.909090909090914,2 57.27272727272727,7.32 63.63636363636363,2 70,2"></polyline></g></svg>
				</td>
				<td>
					<a href="#">Edit</a> / <a href="#">save</a>
				</td>
			</tr>
			<tr>
				<td>yoast seo plugin</td>
				<td>800</td>
				<td class="yoast-table__image">
					<svg width="70" height="30" viewBox="0 0 70 30"><g><polygon fill="#ade3fc" points="0,28 0,12.36 6.363636363636364,12.36 12.727272727272728,7.32 19.09090909090909,12.36 25.454545454545457,7.32 31.818181818181817,7.32 38.18181818181818,7.32 44.54545454545455,2 50.909090909090914,2 57.27272727272727,7.32 63.63636363636363,2 70,2 70,28"></polygon><polyline fill="none" stroke="#498afc" stroke-width="2" points="0,12.36 6.363636363636364,12.36 12.727272727272728,7.32 19.09090909090909,12.36 25.454545454545457,7.32 31.818181818181817,7.32 38.18181818181818,7.32 44.54545454545455,2 50.909090909090914,2 57.27272727272727,7.32 63.63636363636363,2 70,2"></polyline></g></svg>
				</td>
				<td>
					<a href="#">Edit</a> / <a href="#">save</a>
				</td>
			</tr>
			<tr>
				<td>yoast seo plugin</td>
				<td>800</td>
				<td class="yoast-table__image">
					<svg width="70" height="30" viewBox="0 0 70 30"><g><polygon fill="#ade3fc" points="0,28 0,12.36 6.363636363636364,12.36 12.727272727272728,7.32 19.09090909090909,12.36 25.454545454545457,7.32 31.818181818181817,7.32 38.18181818181818,7.32 44.54545454545455,2 50.909090909090914,2 57.27272727272727,7.32 63.63636363636363,2 70,2 70,28"></polygon><polyline fill="none" stroke="#498afc" stroke-width="2" points="0,12.36 6.363636363636364,12.36 12.727272727272728,7.32 19.09090909090909,12.36 25.454545454545457,7.32 31.818181818181817,7.32 38.18181818181818,7.32 44.54545454545455,2 50.909090909090914,2 57.27272727272727,7.32 63.63636363636363,2 70,2"></polyline></g></svg>
				</td>
				<td>
					<a href="#">Edit</a> / <a href="#">save</a>
				</td>
			</tr>
		</tbody>
	</table>

	<p>
		Standard table first column is primary<br />
		Place the <code>yoast-table--primary</code> on the <code>th</code> of the column you want to be primary.
	</p>

	<table class="yoast-table">
		<thead>
			<tr>
				<th class="yoast-table--primary">Related keyphrase</th>
				<th>
					Volume
					<a class="yoast-help" target="_blank" href="https://yoast.com">
						<span class="yoast-help__icon">
							<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 12 12" role="img" aria-hidden="true"
								focusable="false">
								<path
									d="M12 6A6 6 0 110 6a6 6 0 0112 0zM6.2 2C4.8 2 4 2.5 3.3 3.5l.1.4.8.7.4-.1c.5-.5.8-.9 1.4-.9.5 0 1.1.4 1.1.8s-.3.6-.7.9C5.8 5.6 5 6 5 7c0 .2.2.4.3.4h1.4L7 7c0-.8 2-.8 2-2.6C9 3 7.5 2 6.2 2zM6 8a1.1 1.1 0 100 2.2A1.1 1.1 0 006 8z" />
							</svg>
						</span>
						<span class="screen-reader-text">Tell what this link does.</span>
						<span class="screen-reader-text">(Opens in a new browser tab)</span>
					</a>
				</th>
				<th>Trend</th>
				<th>Action</th>
			</tr>
		</thead>
		<tbody>
			<tr>
				<td>yoast seo plugin</td>
				<td>800</td>
				<td class="yoast-table__image">
					<svg width="70" height="30" viewBox="0 0 70 30"><g><polygon fill="#ade3fc" points="0,28 0,12.36 6.363636363636364,12.36 12.727272727272728,7.32 19.09090909090909,12.36 25.454545454545457,7.32 31.818181818181817,7.32 38.18181818181818,7.32 44.54545454545455,2 50.909090909090914,2 57.27272727272727,7.32 63.63636363636363,2 70,2 70,28"></polygon><polyline fill="none" stroke="#498afc" stroke-width="2" points="0,12.36 6.363636363636364,12.36 12.727272727272728,7.32 19.09090909090909,12.36 25.454545454545457,7.32 31.818181818181817,7.32 38.18181818181818,7.32 44.54545454545455,2 50.909090909090914,2 57.27272727272727,7.32 63.63636363636363,2 70,2"></polyline></g></svg>
				</td>
				<td>
					<a href="#">Edit</a> / <a href="#">save</a>
				</td>
			</tr>
			<tr>
				<td>yoast seo plugin</td>
				<td>800</td>
				<td class="yoast-table__image">
					<svg width="70" height="30" viewBox="0 0 70 30"><g><polygon fill="#ade3fc" points="0,28 0,12.36 6.363636363636364,12.36 12.727272727272728,7.32 19.09090909090909,12.36 25.454545454545457,7.32 31.818181818181817,7.32 38.18181818181818,7.32 44.54545454545455,2 50.909090909090914,2 57.27272727272727,7.32 63.63636363636363,2 70,2 70,28"></polygon><polyline fill="none" stroke="#498afc" stroke-width="2" points="0,12.36 6.363636363636364,12.36 12.727272727272728,7.32 19.09090909090909,12.36 25.454545454545457,7.32 31.818181818181817,7.32 38.18181818181818,7.32 44.54545454545455,2 50.909090909090914,2 57.27272727272727,7.32 63.63636363636363,2 70,2"></polyline></g></svg>
				</td>
				<td>
					<a href="#">Edit</a> / <a href="#">save</a>
				</td>
			</tr>
			<tr>
				<td>yoast seo plugin</td>
				<td>800</td>
				<td class="yoast-table__image">
					<svg width="70" height="30" viewBox="0 0 70 30"><g><polygon fill="#ade3fc" points="0,28 0,12.36 6.363636363636364,12.36 12.727272727272728,7.32 19.09090909090909,12.36 25.454545454545457,7.32 31.818181818181817,7.32 38.18181818181818,7.32 44.54545454545455,2 50.909090909090914,2 57.27272727272727,7.32 63.63636363636363,2 70,2 70,28"></polygon><polyline fill="none" stroke="#498afc" stroke-width="2" points="0,12.36 6.363636363636364,12.36 12.727272727272728,7.32 19.09090909090909,12.36 25.454545454545457,7.32 31.818181818181817,7.32 38.18181818181818,7.32 44.54545454545455,2 50.909090909090914,2 57.27272727272727,7.32 63.63636363636363,2 70,2"></polyline></g></svg>
				</td>
				<td>
					<a href="#">Edit</a> / <a href="#">save</a>
				</td>
			</tr>
			<tr>
				<td>yoast seo plugin</td>
				<td>800</td>
				<td class="yoast-table__image">
					<svg width="70" height="30" viewBox="0 0 70 30"><g><polygon fill="#ade3fc" points="0,28 0,12.36 6.363636363636364,12.36 12.727272727272728,7.32 19.09090909090909,12.36 25.454545454545457,7.32 31.818181818181817,7.32 38.18181818181818,7.32 44.54545454545455,2 50.909090909090914,2 57.27272727272727,7.32 63.63636363636363,2 70,2 70,28"></polygon><polyline fill="none" stroke="#498afc" stroke-width="2" points="0,12.36 6.363636363636364,12.36 12.727272727272728,7.32 19.09090909090909,12.36 25.454545454545457,7.32 31.818181818181817,7.32 38.18181818181818,7.32 44.54545454545455,2 50.909090909090914,2 57.27272727272727,7.32 63.63636363636363,2 70,2"></polyline></g></svg>
				</td>
				<td>
					<a href="#">Edit</a> / <a href="#">save</a>
				</td>
			</tr>
		</tbody>
	</table>

	<p>
		Standard table first column is primary and line breaking isn't allowed. <br />
		You can place the <code>yoast-table--nobreak</code> on the <code>table</code>, <code>tr</code> or <code>td</code>.
	</p>

	<table class="yoast-table yoast-table--nobreak">
		<thead>
			<tr>
				<th class="yoast-table--primary">Related keyphrase</th>
				<th>
					Volume
					<a class="yoast-help" target="_blank" href="https://yoast.com">
						<span class="yoast-help__icon">
							<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 12 12" role="img" aria-hidden="true"
								focusable="false">
								<path
									d="M12 6A6 6 0 110 6a6 6 0 0112 0zM6.2 2C4.8 2 4 2.5 3.3 3.5l.1.4.8.7.4-.1c.5-.5.8-.9 1.4-.9.5 0 1.1.4 1.1.8s-.3.6-.7.9C5.8 5.6 5 6 5 7c0 .2.2.4.3.4h1.4L7 7c0-.8 2-.8 2-2.6C9 3 7.5 2 6.2 2zM6 8a1.1 1.1 0 100 2.2A1.1 1.1 0 006 8z" />
							</svg>
						</span>
						<span class="screen-reader-text">Tell what this link does.</span>
						<span class="screen-reader-text">(Opens in a new browser tab)</span>
					</a>
				</th>
				<th>Trend</th>
				<th>Action</th>
			</tr>
		</thead>
		<tbody>
			<tr>
				<td>yoast seo plugin</td>
				<td>800</td>
				<td class="yoast-table__image">
					<svg width="70" height="30" viewBox="0 0 70 30"><g><polygon fill="#ade3fc" points="0,28 0,12.36 6.363636363636364,12.36 12.727272727272728,7.32 19.09090909090909,12.36 25.454545454545457,7.32 31.818181818181817,7.32 38.18181818181818,7.32 44.54545454545455,2 50.909090909090914,2 57.27272727272727,7.32 63.63636363636363,2 70,2 70,28"></polygon><polyline fill="none" stroke="#498afc" stroke-width="2" points="0,12.36 6.363636363636364,12.36 12.727272727272728,7.32 19.09090909090909,12.36 25.454545454545457,7.32 31.818181818181817,7.32 38.18181818181818,7.32 44.54545454545455,2 50.909090909090914,2 57.27272727272727,7.32 63.63636363636363,2 70,2"></polyline></g></svg>
				</td>
				<td>
					<a href="#">Edit</a> / <a href="#">save</a>
				</td>
			</tr>
			<tr>
				<td>yoast seo plugin</td>
				<td>800</td>
				<td class="yoast-table__image">
					<svg width="70" height="30" viewBox="0 0 70 30"><g><polygon fill="#ade3fc" points="0,28 0,12.36 6.363636363636364,12.36 12.727272727272728,7.32 19.09090909090909,12.36 25.454545454545457,7.32 31.818181818181817,7.32 38.18181818181818,7.32 44.54545454545455,2 50.909090909090914,2 57.27272727272727,7.32 63.63636363636363,2 70,2 70,28"></polygon><polyline fill="none" stroke="#498afc" stroke-width="2" points="0,12.36 6.363636363636364,12.36 12.727272727272728,7.32 19.09090909090909,12.36 25.454545454545457,7.32 31.818181818181817,7.32 38.18181818181818,7.32 44.54545454545455,2 50.909090909090914,2 57.27272727272727,7.32 63.63636363636363,2 70,2"></polyline></g></svg>
				</td>
				<td>
					<a href="#">Edit</a> / <a href="#">save</a>
				</td>
			</tr>
			<tr>
				<td>yoast seo plugin</td>
				<td>800</td>
				<td class="yoast-table__image">
					<svg width="70" height="30" viewBox="0 0 70 30"><g><polygon fill="#ade3fc" points="0,28 0,12.36 6.363636363636364,12.36 12.727272727272728,7.32 19.09090909090909,12.36 25.454545454545457,7.32 31.818181818181817,7.32 38.18181818181818,7.32 44.54545454545455,2 50.909090909090914,2 57.27272727272727,7.32 63.63636363636363,2 70,2 70,28"></polygon><polyline fill="none" stroke="#498afc" stroke-width="2" points="0,12.36 6.363636363636364,12.36 12.727272727272728,7.32 19.09090909090909,12.36 25.454545454545457,7.32 31.818181818181817,7.32 38.18181818181818,7.32 44.54545454545455,2 50.909090909090914,2 57.27272727272727,7.32 63.63636363636363,2 70,2"></polyline></g></svg>
				</td>
				<td>
					<a href="#">Edit</a> / <a href="#">save</a>
				</td>
			</tr>
			<tr>
				<td>yoast seo plugin</td>
				<td>800</td>
				<td class="yoast-table__image">
					<svg width="70" height="30" viewBox="0 0 70 30"><g><polygon fill="#ade3fc" points="0,28 0,12.36 6.363636363636364,12.36 12.727272727272728,7.32 19.09090909090909,12.36 25.454545454545457,7.32 31.818181818181817,7.32 38.18181818181818,7.32 44.54545454545455,2 50.909090909090914,2 57.27272727272727,7.32 63.63636363636363,2 70,2 70,28"></polygon><polyline fill="none" stroke="#498afc" stroke-width="2" points="0,12.36 6.363636363636364,12.36 12.727272727272728,7.32 19.09090909090909,12.36 25.454545454545457,7.32 31.818181818181817,7.32 38.18181818181818,7.32 44.54545454545455,2 50.909090909090914,2 57.27272727272727,7.32 63.63636363636363,2 70,2"></polyline></g></svg>
				</td>
				<td>
					<a href="#">Edit</a> / <a href="#">save</a>
				</td>
			</tr>
		</tbody>
	</table>

	<p>
		Standard table first column is primary and line breaking isn't allowed. <br />
		And with buttons styled for a table shown.
	</p>

	<table class="yoast-table yoast-table--nobreak">
		<thead>
			<tr>
				<th class="yoast-table--primary">Related keyphrase</th>
				<th>
					Volume
					<a class="yoast-help" target="_blank" href="https://yoast.com">
						<span class="yoast-help__icon">
							<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 12 12" role="img" aria-hidden="true"
								focusable="false">
								<path
									d="M12 6A6 6 0 110 6a6 6 0 0112 0zM6.2 2C4.8 2 4 2.5 3.3 3.5l.1.4.8.7.4-.1c.5-.5.8-.9 1.4-.9.5 0 1.1.4 1.1.8s-.3.6-.7.9C5.8 5.6 5 6 5 7c0 .2.2.4.3.4h1.4L7 7c0-.8 2-.8 2-2.6C9 3 7.5 2 6.2 2zM6 8a1.1 1.1 0 100 2.2A1.1 1.1 0 006 8z" />
							</svg>
						</span>
						<span class="screen-reader-text">Tell what this link does.</span>
						<span class="screen-reader-text">(Opens in a new browser tab)</span>
					</a>
				</th>
				<th>Trend</th>
				<th>Action</th>
			</tr>
		</thead>
		<tbody>
			<tr>
				<td>yoast seo plugin</td>
				<td>800</td>
				<td class="yoast-table__image">
					<svg width="70" height="30" viewBox="0 0 70 30"><g><polygon fill="#ade3fc" points="0,28 0,12.36 6.363636363636364,12.36 12.727272727272728,7.32 19.09090909090909,12.36 25.454545454545457,7.32 31.818181818181817,7.32 38.18181818181818,7.32 44.54545454545455,2 50.909090909090914,2 57.27272727272727,7.32 63.63636363636363,2 70,2 70,28"></polygon><polyline fill="none" stroke="#498afc" stroke-width="2" points="0,12.36 6.363636363636364,12.36 12.727272727272728,7.32 19.09090909090909,12.36 25.454545454545457,7.32 31.818181818181817,7.32 38.18181818181818,7.32 44.54545454545455,2 50.909090909090914,2 57.27272727272727,7.32 63.63636363636363,2 70,2"></polyline></g></svg>
				</td>
				<td class="yoast-table__button">
					<button class="yoast-button yoast-button--secondary yoast-button--small">Insert key phrase</button>
				</td>
			</tr>
			<tr>
				<td>yoast seo plugin</td>
				<td>800</td>
				<td class="yoast-table__image">
					<svg width="70" height="30" viewBox="0 0 70 30"><g><polygon fill="#ade3fc" points="0,28 0,12.36 6.363636363636364,12.36 12.727272727272728,7.32 19.09090909090909,12.36 25.454545454545457,7.32 31.818181818181817,7.32 38.18181818181818,7.32 44.54545454545455,2 50.909090909090914,2 57.27272727272727,7.32 63.63636363636363,2 70,2 70,28"></polygon><polyline fill="none" stroke="#498afc" stroke-width="2" points="0,12.36 6.363636363636364,12.36 12.727272727272728,7.32 19.09090909090909,12.36 25.454545454545457,7.32 31.818181818181817,7.32 38.18181818181818,7.32 44.54545454545455,2 50.909090909090914,2 57.27272727272727,7.32 63.63636363636363,2 70,2"></polyline></g></svg>
				</td>
				<td class="yoast-table__button">
					<button class="yoast-button yoast-button--secondary yoast-button--small">Insert key phrase</button>
				</td>
			</tr>
			<tr>
				<td>yoast seo plugin</td>
				<td>800</td>
				<td class="yoast-table__image">
					<svg width="70" height="30" viewBox="0 0 70 30"><g><polygon fill="#ade3fc" points="0,28 0,12.36 6.363636363636364,12.36 12.727272727272728,7.32 19.09090909090909,12.36 25.454545454545457,7.32 31.818181818181817,7.32 38.18181818181818,7.32 44.54545454545455,2 50.909090909090914,2 57.27272727272727,7.32 63.63636363636363,2 70,2 70,28"></polygon><polyline fill="none" stroke="#498afc" stroke-width="2" points="0,12.36 6.363636363636364,12.36 12.727272727272728,7.32 19.09090909090909,12.36 25.454545454545457,7.32 31.818181818181817,7.32 38.18181818181818,7.32 44.54545454545455,2 50.909090909090914,2 57.27272727272727,7.32 63.63636363636363,2 70,2"></polyline></g></svg>
				</td>
				<td class="yoast-table__button">
					<button class="yoast-button yoast-button--secondary yoast-button--small">Insert key phrase</button>
				</td>
			</tr>
			<tr>
				<td>yoast seo plugin</td>
				<td>800</td>
				<td class="yoast-table__image">
					<svg width="70" height="30" viewBox="0 0 70 30"><g><polygon fill="#ade3fc" points="0,28 0,12.36 6.363636363636364,12.36 12.727272727272728,7.32 19.09090909090909,12.36 25.454545454545457,7.32 31.818181818181817,7.32 38.18181818181818,7.32 44.54545454545455,2 50.909090909090914,2 57.27272727272727,7.32 63.63636363636363,2 70,2 70,28"></polygon><polyline fill="none" stroke="#498afc" stroke-width="2" points="0,12.36 6.363636363636364,12.36 12.727272727272728,7.32 19.09090909090909,12.36 25.454545454545457,7.32 31.818181818181817,7.32 38.18181818181818,7.32 44.54545454545455,2 50.909090909090914,2 57.27272727272727,7.32 63.63636363636363,2 70,2"></polyline></g></svg>
				</td>
				<td class="yoast-table__button">
					<button class="yoast-button yoast-button--secondary yoast-button--small">Insert key phrase</button>
				</td>
			</tr>
		</tbody>
	</table>
</body>

</html>
