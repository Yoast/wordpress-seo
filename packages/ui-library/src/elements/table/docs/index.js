export { default as component } from "./component.md";
export { default as tableBody } from "./table-body.md";
export { default as tableCell } from "./table-cell.md";
export { default as tableHead } from "./table-head.md";
export { default as tableHeader } from "./table-header.md";
export { default as tableRow } from "./table-row.md";
export { default as minimal } from "./table-variant-minimal.md";
