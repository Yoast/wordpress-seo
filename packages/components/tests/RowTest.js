import React from "react";
import renderer from "react-test-renderer";

import { Row, RowResponsiveWrap } from "../src/table/Row";

test( "the Row component matches the snapshot", () => {
	const component = renderer.create(
		<Row rowHeight="48px" />
	);

	const tree = component.toJSON();
	expect( tree ).toMatchSnapshot();
} );

test( "the RowResponsiveWrap component matches the snapshot", () => {
	const component = renderer.create(
		<RowResponsiveWrap />
	);

	const tree = component.toJSON();
	expect( tree ).toMatchSnapshot();
} );

