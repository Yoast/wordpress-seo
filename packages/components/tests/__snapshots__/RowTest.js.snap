// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the Row component matches the snapshot 1`] = `
.c0 {
  background: #fff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  min-height: 48px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

<li
  className="c0"
/>
`;

exports[`the RowResponsiveWrap component matches the snapshot 1`] = `
.c0 {
  background: #fff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

@media screen and ( max-width:800px ) {
  .c1 {
    -webkit-flex-wrap: wrap;
    -ms-flex-wrap: wrap;
    flex-wrap: wrap;
    -webkit-align-items: flex-start;
    -webkit-box-align: flex-start;
    -ms-flex-align: flex-start;
    align-items: flex-start;
  }

  .c1:first-child {
    margin-top: 24px;
  }

  .c1 > span::before {
    position: static;
    display: inline-block;
    padding-right: 0.5em;
    font-size: inherit;
  }

  .c1 > span {
    padding-left: 0;
  }
}

<li
  className="c0 c1"
/>
`;
